external/js/kyber/lib/sign/schnorr/schnorr.js

```diff
@@ -0,0 +1,128 @@
+const group = require("./../../group");
+// XXX to be changed to an interface style
+const nist = group.nist;      
+const curve = nist.Curve;
+const scalarCons = nist.Scalar;
+const pointCons = nist.Point;
+
+const hash = require("hash.js");
+/**
+ *
+ * Sign computes a Schnorr signature over the given message.
+ * 
+ * @param privateKey private key scalar to sign with
+ * @param message message over which the signature is computed
+ * @return signature as a Uint8Array
+ * */
+function Sign(suite, privateKey, message) {
+    if (suite.constructor !== curve) {
+        throw "first argument must be a suite";
+    }
+    if (privateKey.constructor !== scalarCons) {
+        throw "second argument must be a scalar";
+    }
+    if (message.constructor !== Uint8Array) {
+        throw "third argument must be Uint8Array";
+    }
+
+    // generate r & R
+    const r = suite.scalar().pick();
+    const R = suite.point().mul(r,null);
+    const buffR = R.marshalBinary();
+
+    // generate public key
+    const pub = suite.point().mul(privateKey,null);
+    
+    // generate challenge
+    const challenge = hashSchnorr(suite,buffR, pub.marshalBinary(), 
+                         message) 
+
+    // generate signature
+    const s = suite.scalar().mul(privateKey,challenge)
+    s.add(s,r)
+
+    // concatenate R || s
+    const buffS = s.marshalBinary();
+    const buffSig = new Uint8Array(buffR.length + buffS.length);
+    buffSig.set(buffR);
```

Comment:
I'm a JS-noob: how can you set a `const` to another value? Should `const buffSig` with `let buffSig`?

Reply:
I don't understand. I'm also not a JS expert, but it's the only way I know of filling a buffer. I can use let probably but why should I, if it's meant to be a constant anyway ? (i.e. the variable is meant to always be this `Uint8Array` but the content of the array may change)

Reply:
Looking at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const it seems that this should be replaced with `let buffSig`.

Reply:
Why ? 
> The const declaration creates a read-only reference to a value. It does not mean the value it holds is immutable, just that the variable identifier cannot be reassigned. For instance, in the case where the content is an object, this means the object's contents (e.g., its parameters) can be altered.
So it's exactly this case. What's wrong ?

Reply:
OK - I didn't see this on the link from above. Where do you read this?

Reply:
Second paragraph of description !